### ISSUE TITLE
Windows 10 / 8 Start Menu Tiles

### ISSUE THREAD

**Issue by hyarsan** [libretro/RetroArch#6000]
I was working on Windows 10 / 8 start menu tiles for RetroArch:

![pretty_cool](https://user-images.githubusercontent.com/32988993/34454507-3cae572e-ed6d-11e7-96f1-0751fc300452.png)
( These are all examples of desktop apps that have fancy tiles, except for the 3 tiles at the top )
( They also have small variants :D )

I got it working, but.... where should the tile graphics and the manifest file for it be stored?
Should the tile graphics be in retroarch-assets, and have the start menu manifest compile with RetroArch Windows? ( I don't yet know how to do that, though. )
And the start menu manifest files need to be in the root directory of RetroArch...

**Comment by bparker06** [comment]
Can you explain how this works on a non-UWP app? If there is a way, first we'd need to know what the requirements are for compiling it in (if needed) or getting it to work in general before even discussing where things may go.

**Comment by hyarsan** [comment]
In order to have fancy start menu tiles for non-UWP apps, you need to have an start menu manifest file in the same directory as the executable that you want to pin to Start. In this case, I have a "retroarch.VisualElementsManifest.xml" and a "retroarch_debug.VisualElementsManifest.xml".
The manifest file tells the start menu where the tile images are what the color is and stuff like that.

Now here is what I want to know: How would the manifest compile with RetroArch and where should the tile images be located at? In a "startmenu" folder inside "assets"?

**Comment by bparker06** [comment]
I thought you just said the manifest has to be in the same directory? Wouldn't that imply that it would not be compiled in?

**Comment by bparker06** [comment]
BTW [this](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/appxmanifestschema/element-visualelements) doesn't mention anything about non-UWP... so I originally assumed that it wasn't possible. But if that's not the case, I guess that's a good thing.

**Comment by hyarsan** [comment]
Yeah it's cool. The reason I thought that is because at what other location can it be? I thought otherwise it would need to be in assets/ which obviously won't work.

**Comment by hyarsan** [comment]
Oh and another thing I should note is, in order to have it say the fancy names "RetroArch" and "RetroArch Debug" you would need to pin a shortcut of the executable, not the executable itself. So can we have a "RetroArch Debug" shortcut come with the Windows installer? Thanks

**Comment by hyarsan** [comment]
With compiled in I meant "have it come with the installer / have it be in the RetroArch directory when you download it or **compile** it"

**Comment by bparker06** [comment]
Can you give an example manifest that works? I tried but wasn't able to get anything to appear.

**Comment by hyarsan** [comment]
You can just install Chrome or Firefox and pin those but here I guess:

https://1drv.ms/u/s!AvB-c2zUfIal2jXpJ_e7o1g949zR

Extract it and drag all of it's contents to the RetroArch directory. I only included the regular RetroArch one.

**Comment by bparker06** [comment]
Thanks, that one works... although the parameters given are totally different from what the link I pasted shows is even possible, so I'm a little confused.

The manifest file and the icon(s) should probably just be added to the installer where it would put them in the root folder of the installation, next to the exe.

**Comment by hyarsan** [comment]
Yeah, it was pretty hard to find proper documentation on it :P Who's actually going to implement it? I'll give you the rest after I get sleep.

**Comment by hyarsan** [comment]
Are you sure that just throwing 6 files for the start menu to look nice next to the executable is a good idea? Anyways hey @WhoeverWorksOnAddingThingsToTheInstaller423426, can you add these files next to the executable and add a shortcut that's named "RetroArch" and "RetroArch Debug"?
https://1drv.ms/u/s!AvB-c2zUfIal2jayQuxZF8OF_3QU

**Comment by bparker06** [comment]
I wrote the installer so it'd be me adding them... but we need @twinaphex's opinion on this before doing anything. I don't think it will hurt to have a few extra small files next to the exe... but at the same time, none of this is visible to the end user unless they happen to pin RA to the Start Menu.

**Comment by hyarsan** [comment]
I am not sure why, but sometimes the tile icons for regular RetroArch's shortcut doesn't work?

**Comment by twinaphex** [comment]
@bparker06 Generally I am not opposed to it, if it can make RetroArch look and feel like a regular program then it's a plus in my book. As long as we can keep this to the installer.

**Comment by hyarsan** [comment]
I also guess this is good preparation for RetroArch UWP :P

**Comment by hyarsan** [comment]
https://twitter.com/YarsanHoessain/status/948976126554263552
I tweeted this :P

**Comment by orbea** [comment]
@hyarsan and @bparker06 Was this ever implemented?

**Comment by hyarsan** [comment]
nope.

**Comment by hyarsan** [comment]
This would especially be nice for RetroArch UWP.

**Comment by amirrammari** [comment]
@hyarsan Very Nice +1
@orbea Please implement this.
![RetroArch Windows 10 Tile](https://user-images.githubusercontent.com/23485114/68546806-d7c06a00-03ef-11ea-9397-971ed48f0d74.png)
